internal/pkg/workspace/workspace.go

```diff
@@ -151,6 +152,52 @@ func (ws *Workspace) ListWorkloads() ([]string, error) {
 	})
 }
 
+// ListPipelines returns the names of all pipelines in the workspace.
+func (ws *Workspace) ListPipelines() (map[string]string, error) {
+	pipelines := make(map[string]string)
+	// Look for legacy pipeline.
+	legacyPath, err := ws.PipelineManifestLegacyPath()
+	if err != nil {
+		return nil, err
+	}
+	name, err := ws.PipelineNameFromManifest(legacyPath)
+	if err != nil {
+		return nil, fmt.Errorf("get name from pipeline manifest: %w", err)
+	}
+	pipelines[name] = legacyPath
+	// Look for other pipelines.
+	pipelinesPath, err := ws.pipelineManifestPath()
+	if err != nil {
+		return nil, err
+	}
+	exists, err := ws.fsUtils.Exists(pipelinesPath)
+	if err != nil {
+		return nil, err
+	}
+	if exists {
+		files, err := ws.fsUtils.ReadDir(pipelinesPath)
+		if err != nil {
+			return nil, fmt.Errorf("read directory %s: %w", pipelinesPath, err)
+		}
+		for _, file := range files {
+			// Ignore buildspecs.
+			if strings.HasSuffix(file.Name(), "buildspec.yml") {
+				continue
+			}
+			// Read manifests of moved legacy pipeline and any other pipelines.
+			if strings.HasSuffix(file.Name(), ".yml") {
+				path := filepath.Join(pipelinesPath, file.Name())
+				name, err := ws.PipelineNameFromManifest(path)
+				if err != nil {
+					return nil, fmt.Errorf("read pipeline manifest: %w", err)
+				}
```

Comment:
what do you think of not erroring but instead maybe `log.Info` this and skipping the file and `continue`ing? this way we can fetch as many pipeline paths as we can without stopping